src/opustrainer/trainer.py

```diff
@@ -394,18 +497,24 @@ def _load_modifiers(self, ymldata:dict) -> List[Modifier]:
         """Reads
         ```yml
         modifiers:
-          - uppercase 0.05
-          - titlecase 0.05
+          - uppercase:
+            - probability: 0.05
+          - titlecase:
+            - probability: 0.05
```

Comment:
Why the double list + dict notation here? What's against:
```yaml
modifiers:
- uppercase:
    probability: 0.5
- titlecase:
    probability: 0.5
```

Since all filters have to have a probability anyway, you could keep it even closer to the original syntax by taking the value of that first key to be probability:
```yaml
modifiers:
- uppercase: 0.5
- tags: 0.08
  num_tags: 6
  chinese: src
```

The extra indentation of the dict is even optional, yaml is really flexible with that apparently. Also it matches closer to [what OpusFilter has been doing](https://github.com/Helsinki-NLP/OpusFilter/blob/develop/example_configs/qed_lm_langid.yaml). We could even follow them and treat the modifier names as class names, and use TitleCase for them.

```yaml
modifiers:
- UpperCase: 0.5
- Tags: 0.08
  num_tags: 6
  chinese: src
```

that has the added benefit that you could even have a `Tags` filter with a `tags` parameter.

Reply:
Fixed.